hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
-# copy_digest_tag "<PULL_SECRET>" "src_acr_name" "dst_acr_name" "$(get_digest_tag FluentbitImage)"
+skip_and_import_certificates(){
+    err_str="Usage $0 <KEYVAULT>  <CERTIFICATEs...> <SECRET_FILES...> [SKIP_DEPLOYMENTS]. Please try again"
+    local keyVault=${1?$err_str}
+
+    # Extract the certificates and secret_files arraies based on half the remaining arguments
+    local num_certs=$((($# -1) / 2))
+    local -a certificates=("${@:2:$num_certs}")
+    local -a secret_files=("${@:$((num_certs + 2)):$num_certs}")
+    # Optional flag (last argument)
+    local skip_deployments="${@: -1}"
+
+    # Don't skip deployment creation when skip_deployments was set to "false" 
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        abort "'skip_deployments' was set to 'false'. ❌⏩ Don't skip certs import to keyVault $keyVault"
+    fi
+
+    for i in "${!certificates[@]}"; do
+         if check_keyvault_certificate "${keyVault}" "${certificates[i]}" $skip_deployments; then
+            log "⏭️🔑💼 Skip import for certificate ${certificates[i]}"
+        else
+            log "🔐📥 Import certificate ${certificates[i]}"
+            az keyvault certificate import \
+                --vault-name "${keyVault}" \
+                --name "${certificates[i]}" \
+                --file "${secret_files[i]}" >/dev/null
+        fi
+    done
+}
+
+check_and_import_certificates (){
+    err_str="Usage $0 <KEYVAULT_PREFIX> [SKIP_DEPLOYMENTS]. Please try again"
+    local keyvault_prefix=${1?$err_str}
+    local skip_deployments=${2:-"true"}
+
+    # Don't skip deployment creation when skip_deployments was set to "false" 
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        abort "'skip_deployments' was set to 'false'. ❌⏩ Don't skip certs import"
+    fi
+
+    local files
+    local certs
+    local key_vault
+    key_vault="${keyvault_prefix}-svc"
+    log "Check import certificates for the service keyVault ${key_vault}"
+    certs=("rp-mdm" "rp-mdsd" "cluster-mdsd" "dev-arm" "rp-firstparty" "rp-server")
+    files=("secrets/rp-mdm-self-signed.pem" "secrets/rp-mdsd-self-signed.pem" "secrets/cluster-mdsd-self-signed.pem" "secrets/arm.pem" "secrets/firstparty.pem" "secrets/localhost.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+    
+    key_vault="${keyvault_prefix}-gwy"
+    log "Check import certificates for the gateway keyVault ${key_vault}"
+    certs=("gwy-mdm" "gwy-mdsd")
+    files=("secrets/gwy-mdm-self-signed.pem" "secrets/gwy-mdsd-self-signed.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+
+    key_vault="${keyvault_prefix}-por"
+    log "Check import certificates for the portal keyVault ${key_vault}"
+    certs=("portal-server" "portal-client")
+    files=("secrets/localhost.pem" "secrets/portal-client.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
```

Comment:
Double quote to prevent re-splitting array elements.

```suggestion
    skip_and_import_certificates "${key_vault}" "${certs[@]}" "${files[@]}" "$skip_deployments"
    
    key_vault="${keyvault_prefix}-gwy"
    log "Check import certificates for the gateway keyVault ${key_vault}"
    certs=("gwy-mdm" "gwy-mdsd")
    files=("secrets/gwy-mdm-self-signed.pem" "secrets/gwy-mdsd-self-signed.pem")
    skip_and_import_certificates "${key_vault}" "${certs[@]}" "${files[@]}" "$skip_deployments"

    key_vault="${keyvault_prefix}-por"
    log "Check import certificates for the portal keyVault ${key_vault}"
    certs=("portal-server" "portal-client")
    files=("secrets/localhost.pem" "secrets/portal-client.pem")
    skip_and_import_certificates "${key_vault}" "${certs[@]}" "${files[@]}" "$skip_deployments"
```